test/libsolidity/smtCheckerTests/loops/for_loop_bmc_iterations_semantics_3.sol

```diff
@@ -6,14 +6,14 @@ contract C {
 			if (x >= 3)
 				y = 1;
 		}
-        // BMC loop iteration setting is more than enough to leave the loop
+		// BMC loop iteration setting is more than enough to leave the loop
 		assert(x == 3); // should hold - no assumptions on value if didn't complete loop
 		assert(y == 0); // should hold - no assumptions on value if didn't complete loop
 	}
 }
 // ====
 // SMTEngine: bmc
 // SMTSolvers: z3
-// BMCLoopIterations: 5
+// BMCLoopIterations: 4
```

Comment:
Why are this one and a few other tests decreased?

Reply:
Because the semantics changed a bit.
I think it's better when loop iterations settings works like this:
- go through the loop `--model-checker-bmc-loop-iterations` number of times (including the loop body)
- in case of `while` and `for` loops visit the condition again - final condition check like `while` and `for` are supposed to work

Previously there was no final check of those conditions.

According to that change the numbers decreased.
@leonardoalt @blishko

Reply:
But this should still give the same results with 5, shouldn't it?

Reply:
right, I could leave it like it was, but I wanted to be consistent with what we had previously:
- 1 test for "not enough"
- 1 test for "just enough"
- 1 test for "more than enough" which was just enough + 1

Reply:
ok makes sense